app/Http/Controllers/UserController.php

```diff
@@ -128,19 +141,53 @@ public function check(Request $request)
                 $score=$score+1;
             }
         }
-        
-        $collection = collect();
 
-        for($j=0;$j<$question->count();$j++){
-            $collection->push($question[$j]->term);
-        }
-        
+        $collection = $question->map(function ($item,$key){
+            return $item->term;
+        });
+
         $collection1 = collect();
-        $answer= Answer::where('lesson_id','=',$lessonId)->get();
-        for($k=0;$k<$question->count();$k++){
-            $choice = Choices::where('id','=',(int)$answer[$k]->choice_id)->first();
-            $collection1->push($choice);
-        }
+
+        $collection1 = $answers->map(function($item,$key){
+            $choice = Choices::where('id','=',(int)$item->choice_id)->first();
+            return $choice;
+        });
+
         return view('users.check',compact('score','question','answers','choices','collection','collection1','categoryId','lessonId'));
     }
+
+    public function wordsLearned()
+    {
```

Comment:
In this function instead getting data in controllers, create getters and setters function in your Model class and call it in your Controller